### PR TITLE
Only install `.host.a` libraries on non-SGX platforms

### PR DESCRIPTION
**Pull request by jumaffre**
Part of #4419 

While we still need to build virtual/host variant of the libraries to be able to build unit tests for the SGX compile target, we do not need to install these in the SGX Debian package. This should reduce the size of the SGX Debian package for 3.0.0.

### CODE REVIEW

File: cmake/common.cmake [hunk 1]
```diff
@@ -312,13 +312,16 @@ if(COMPILE_TARGET STREQUAL "sgx")
...
 endif()
+
 add_library(http_parser.host "${HTTP_PARSER_SOURCES}")
 set_property(TARGET http_parser.host PROPERTY POSITION_INDEPENDENT_CODE ON)
-install(
-  TARGETS http_parser.host
-  EXPORT ccf
-  DESTINATION lib
-)
+if(NOT COMPILE_TARGET STREQUAL "sgx")
```
Comment (DomAyre) [inline cmake/common.cmake:318]: Would it be more precise to state which platforms do need this rather than which ones don't?

Comment (achamayou) [reply cmake/common.cmake:318]: I'd also rather have a INSTALL_HOST_LIBRARIES, set in a single place if we can.

File: cmake/common.cmake [hunk 2]
```diff
@@ -336,12 +339,14 @@ if(COMPILE_TARGET STREQUAL "sgx")
...
 add_host_library(ccf_kv.host "${CCF_KV_SOURCES}")
 add_san(ccf_kv.host)
-add_warning_checks(ccf_kv.host)
-install(
-  TARGETS ccf_kv.host
-  EXPORT ccf
-  DESTINATION lib
-)
+if(INSTALL_VIRTUAL_LIBRARIES)
+  add_warning_checks(ccf_kv.host)
```
Comment (jumaffre) [inline cmake/common.cmake:343]:
```suggestion
add_warning_checks(ccf_kv.host)
if(INSTALL_VIRTUAL_LIBRARIES)
```

**Review by achamayou — APPROVED** [review]

**Comment by plietar** [comment]
Having these libraries around was useful to write unit tests in downstream applications.
At the very least, we would want to have `ccfcrypto`, `t_cose` and `qcbor`.

`ccf_endpoints` would be nice but I can probably work around it with some refactoring (it isn't actually run by the tests, but I have files where part of the file is tested, and other parts depend on it).

**Pull request merged**